includes/class-wc-subscription-data-copier.php

```diff
@@ -0,0 +1,463 @@
+	 * operational, address and internal meta data is passed through filters before it is copied.
+	 */
+	public function copy_data() {
+
+		if ( $this->is_cpt_data_store() ) {
+			$data_array = $GLOBALS['wpdb']->get_results( $this->get_deprecated_meta_query(), ARRAY_A );
+			$data       = wp_list_pluck( $data_array, 'meta_value', 'meta_key' );
+		} else {
+			$data  = $this->get_meta_data();
+			$data += $this->get_order_data();
+			$data += $this->get_operational_data();
+			$data += $this->get_address_data();
+
+			// Remove any excluded meta keys.
+			$data = $this->filter_excluded_meta_keys_via_query( $data );
+		}
+
+		$data = $this->apply_deprecated_filter( $data );
+		$data = apply_filters( "wc_subscriptions_{$this->copy_type}_data", $data, $this->to_object, $this->from_object );
+		$data = apply_filters( 'wc_subscriptions_copied_data', $data, $this->to_object, $this->from_object, $this->copy_type );
+
+		foreach ( $data_array as $object_data ) {
+			$this->set_data( $object_data['meta_key'], $object_data['meta_value'] );
+		}
+
+		$this->to_object->save();
+	}
+
+	/**
+	 * Sets a piece of data on the "to" object.
+	 *
+	 * This function uses a setter where appropriate, otherwise it sets the data directly.
+	 * Values which are stored as a bool in memory are converted before being set. eg 'no' -> false, 'yes' -> true.
+	 *
+	 * @param string $key   The data key to set.
+	 * @param mixed  $value The value to set.
+	 */
+	private function set_data( $key, $value ) {
+
+		// WC will automatically set/update these keys when a shipping/billing address attribute changes so we can ignore these keys.
+		if ( in_array( $key, array( '_shipping_address_index', '_billing_address_index' ), true ) ) {
+			return;
+		}
+
+		// Special cases where properties with setters don't map nicely to their function names.
+		$setter_map = array(
+			'_cart_discount'      => 'set_discount_total',
+			'_cart_discount_tax'  => 'set_discount_tax',
+			'_customer_user'      => 'set_customer_id',
+			'_order_tax'          => 'set_cart_tax',
+			'_order_shipping'     => 'set_shipping_total',
+			'_order_currency'     => 'set_currency',
+			'_order_shipping_tax' => 'set_shipping_tax',
+			'_order_total'        => 'set_total',
+			'_order_version'      => 'set_version',
+		);
+
+		$setter = isset( $setter_map[ $key ] ) ? $setter_map[ $key ] : 'set_' . ltrim( $key, '_' );
+
+		if ( is_callable( array( $this->to_object, $setter ) ) ) {
+			// Re-bool the value before setting it. Setters like `set_prices_include_tax()` expect a bool.
+			if ( is_string( $value ) && in_array( $value, array( 'yes', 'no' ), true ) ) {
+				$value = 'yes' === $value;
+			}
+
+			$this->to_object->{$setter}( $value );
+		} else {
+			$this->to_object->update_meta_data( $key, $value );
+		}
+	}
+
+	/**
+	 * Determines if the store is using the WP CPT datastores.
+	 *
+	 * @return bool
+	 */
+	private function is_cpt_data_store() {
+		return true;
```

Comment:
This function can be removed and can use wcs_is_custom_order_tables_usage_enabled() instead once #224 is merged.

Reply:
```suggestion
		return function_exists( 'wcs_is_custom_order_tables_usage_enabled' ) ? ! wcs_is_custom_order_tables_usage_enabled() : true;
```

Were you going to keep this function around once #224 is merged or replace the `is_using_cpt_data_stores()` usage inside `copy_data()` and then delete this function?

If we're keeping it around, we could replace it with this suggestion so that we don't have to come back and update this function.

Reply:
> Were you going to keep this function around once https://github.com/Automattic/woocommerce-subscriptions-core/pull/224 is merged or replace the is_using_cpt_data_stores() usage inside copy_data() and then delete this function?

I was going to do this. I don't think there's any need for this function to remain as a wrapper.